Build on QNX, as one of the BSD-style systems.

constants_dep.h stops with "#error PTPD has not been ported to this OS" on
anything but Linux, the BSDs, macOS and Solaris. QNX 8 has the BSD network
API (io-sock is FreeBSD-derived): <net/if_dl.h>, getifaddrs(), IF_NAMESIZE,
struct ether_addr with "octet", so it takes the BSD branch of the platform
constants.

Upstream-Status: QNX-specific

--- a/src/dep/constants_dep.h
+++ b/src/dep/constants_dep.h
@@ -16,7 +16,8 @@
 /* platform dependent */
 
 #if !defined(linux) && !defined(__NetBSD__) && !defined(__FreeBSD__) && \
-  !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__sun)
+  !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__sun) && \
+  !defined(__QNX__)
 #error PTPD hasn't been ported to this OS - should be possible \
 if it's POSIX compatible, if you succeed, report it to ptpd-devel@sourceforge.net
 #endif
@@ -34,7 +35,7 @@
 #define octet ether_addr_octet
 #endif /* linux */
 
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__) || defined(__OpenBSD__) || defined(__sun)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__) || defined(__OpenBSD__) || defined(__sun) || defined(__QNX__)
 # include <sys/types.h>
 # include <sys/socket.h>
 #ifdef HAVE_SYS_SOCKIO_H
